### ISSUE TITLE
ENTK: Task manager error ?

### ISSUE THREAD

**Issue by ValHayot** [radical-collaboration/Glatard-Lab#2]
Hi,

I followed the documentation and have created a simple pipeline (can be found here: [inc.py](https://github.com/ValHayot/Glatard-Lab/tree/master/sample_pipeline/inc.py)). I'm trying to run it on XSEDE, and the pilots are definitely being scheduled according to `squeue`, but it's not producing the output I'm looking for, and I think it may be due to the task manager error seen below:

This is RP stdout:
```
EnTK session: re.session.js-156-107.jetstream-cloud.org.vhayot.018302.0001
Creating AppManagerSetting up RabbitMQ system                                 ok
                                                                              ok
Validating and assigning resource manager                                     ok
Setting up RabbitMQ system                                                   n/a
new session: [re.session.js-156-107.jetstream-cloud.org.vhayot.018302.0001]    \
database   : [mongodb://149.165.156.107:27017/radicaldb]
        ok
create pilot manager                                                          ok
submit 1 pilot(s)
        [xsede.bridges:5]
                                                                              ok
All components created
Update: pipeline.0000 state: SCHEDULING
Update: pipeline.0000.stage.0000 state: SCHEDULING
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULING
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULING
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULING
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULING
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULING
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULED
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULED
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULED
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULED
Update: pipeline.0000.stage.0000.Incrementation 0 state: SCHEDULED
Update: pipeline.0000.stage.0000 state: SCHEDULED
create unit manager/home/vhayot/venv/lib/python3.5/site-packages/pymongo/topology.py:155: UserWarning: MongoClient opened before fork. Create MongoClient only after forking. See PyMongo's documentation for details: http://api.mongodb.org/python/current/faq.html#is-pymongo-fork-safe
  "MongoClient opened before fork. Create MongoClient only "
                                                           ok
add 1 pilot(s)                                                                ok
close unit manager                                                            ok
Process task-manager:
Traceback (most recent call last):
  File "/home/vhayot/venv/lib/python3.5/site-packages/radical/entk/execman/rp/task_manager.py", line 172, in _tmgr
    body = json.loads(body)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/vhayot/venv/lib/python3.5/site-packages/radical/entk/execman/rp/task_manager.py", line 195, in _tmgr
    raise EnTKError(e)
radical.entk.exceptions.EnTKError: the JSON object must be str, not 'bytes'
create unit manager/home/vhayot/venv/lib/python3.5/site-packages/pymongo/topology.py:155: UserWarning: MongoClient opened before fork. Create MongoClient only after forking. See PyMongo's documentation for details: http://api.mongodb.org/python/current/faq.html#is-pymongo-fork-safe
  "MongoClient opened before fork. Create MongoClient only "
                                                           ok
add 1 pilot(s)                                                                ok
close unit manager                                                            ok
<repeats a  few times>
wait for 1 pilot(s)
              0                                                               ok
closing session re.session.js-156-107.jetstream-cloud.org.vhayot.018302.0001   \
close pilot manager                                                            \
wait for 1 pilot(s)
              0                                                          timeout
                                                                              ok
session lifetime: 1128.3s                                                     ok
All components terminated
```
Here are the pilot logs:
[pilot.0000.tar.gz](https://github.com/radical-collaboration/Glatard-Lab/files/4178547/pilot.0000.tar.gz)

I tried running it with `/bin/echo` as well just to make sure it wasn't the application within the task that was bugging, and I got the same error. I can't seem to find anything relevant in the pilot logs other than the usual `libcloud module not found`. The fact that `unit.0000` is also missing from my logs seems to further support the idea that the task was never launched on the pilot. 

Do I have to update a library? Is it my python version?

**Comment by ValHayot** [comment]
this appears to be a duplicate of [#411](https://github.com/radical-cybertools/radical.entk/issues/411)
